db/crud.go

```diff
@@ -583,10 +589,12 @@ func (db *Database) PutExistingRev(docid string, body Body, docHistory []string)
 
 		// Add all the new-to-me revisions to the rev tree:
 		for i := currentRevIndex - 1; i >= 0; i-- {
-			doc.History.addRevision(RevInfo{
+			if err := doc.History.addRevision(RevInfo{
 				ID:      docHistory[i],
 				Parent:  parent,
-				Deleted: (i == 0 && deleted)})
+				Deleted: (i == 0 && deleted)}); err != nil {
+				return nil, nil, err
+			}
```

Comment:
This is a bit difficult to read - it looks like the return on error is a RevInfo field.  I think a standalone if statement might be more straightforward.